DOCS-1197: Review modular resource docs (#2106)
viamrobotics/docs: docs/modular-resources/key-concepts.md

When the module initializes, it registers those pairs on your robot, making the functionality defined by that pair available for use.

## Resources

A resource is a [component](/components/) or [service](/services/).
Each component or service is typed by a proto API, such as the [component proto definitions](https://github.com/viamrobotics/api/tree/main/proto/viam/component).

Any resource on your robot needs to implement either one of the [existing Viam APIs](#valid-apis-to-implement-in-your-model), or a [custom interface](/modular-resources/advanced/#new-api-subtypes).

A _modular resource_ is a resource that is provided by a [module](#modules), and not built into the RDK.
A modular resource runs in the module process.
This differs from built-in resources, which run as part of `viam-server`.

## Models

A _model_ describes a specific implementation of a [resource](#resources) that implements (speaks) its [API](/program/apis/).

> Models allow you to control hardware or software of a similar category (such as motors) with a consistent interface (set of methods), even if the underlying implementation differs.

Reviewer: ```suggestion
Models allow you to control hardware or software of a similar category, such as motors, with a consistent set of methods as an interface, even if the underlying implementation differs.
```

> Models allow you to control hardware or software of a similar category, such as motors, with a consistent set of methods as an interface, even if the underlying implementation differs.

Reviewer 2: "with a consistent set of methods as an interface" sounds like it's missing a direct object. Is there a way to split this sentence to help clarify what we mean?

Author: ```suggestion
Models allow you to use the same set of methods to control hardware or software of a similar category, even if the underlying implementation differs.
```
Is this better?

Reviewer 2: Perfect!

  }
  ```

> The `viam` namespace is reserved for models provided by Viam.

Reviewer 2: Much better, thank you!
